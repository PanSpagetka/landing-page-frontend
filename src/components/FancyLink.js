import React from 'react';
import propTypes from 'prop-types';
import classnames from 'classnames';

import { ArrowRightIcon } from '@patternfly/react-icons';

import './FancyLink.scss';

const FancyLink = ({ to, children, className, ...props }) => {
    return (
        <a className={ classnames('ins-c-link__fancy', className) } href={ to } { ...props }>{ children }<ArrowRightIcon/></a>
    );
};

export default FancyLink;

FancyLink.propTypes = {
    to: propTypes.string,
    className: propTypes.string,
    children: propTypes.any
};
