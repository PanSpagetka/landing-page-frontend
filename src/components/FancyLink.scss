.ins-c-link__fancy {

    --ins-c-link__fancy-red: #E80000; // This is a brand color

    display: grid;
    grid-template-columns: auto 1fr;
    column-gap: var(--pf-global--spacer--md);
    word-break: break-word;

    color: var(--pf-global--Color--100);

    border-left: 2px solid var(--ins-c-link__fancy-red);
    padding-left: var(--pf-global--spacer--sm);

    &:hover, &:hover * {
        text-decoration: none;
        font-weight: var(--pf-global--FontWeight--redhatfont--bold);
        color: var(--pf-global--Color--100);
    }

    svg {
        margin-top: var(--pf-global--spacer--xs);
        color: var(--pf-global--Color--light-300);
    }
}