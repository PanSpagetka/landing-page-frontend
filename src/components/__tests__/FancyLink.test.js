import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FancyLink from '../FancyLink';

describe('BannerCard component', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<FancyLink to='#' className='testClassname'>Test</FancyLink>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
