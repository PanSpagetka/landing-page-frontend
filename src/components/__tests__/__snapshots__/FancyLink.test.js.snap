// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerCard component should render correctly 1`] = `
<a
  className="ins-c-link__fancy testClassname"
  href="#"
>
  Test
  <ArrowRightIcon
    color="currentColor"
    noVerticalAlign={false}
    size="sm"
    title={null}
  />
</a>
`;
